### PR TITLE
Adds support for channel binding for http-SSPI-Negotiate.

### PR DESCRIPTION
**Pull request by georgeok**
Attempt to add support for Secure Channel binding when negotiate authentication is used. The problem to solve is that by default IIS accepts channel binding and curl doesn't utilise them. The result was a 401 response. Scope affects only the Schannel(winssl)-SSPI combination.

### CODE REVIEW

**Comment by georgeok** [comment]
@MarcelRaad  Since you've seen the same patch for NTLM, could you please help me out with this PR? One of the tests is failing in `buildbot/curl_winssl_mingw_w64_x64_dbg`. Looks like a networking issue to me. Any comments?

File: lib/urldata.h [hunk 1]
```diff
@@ -980,6 +983,9 @@ struct connectdata {
   void *seek_client;            /* pointer to pass to the seek() above */
 
   /*************** Request - specific items ************/
+#ifdef SECPKG_ATTR_ENDPOINT_BINDINGS
```
Comment (MarcelRaad) [inline lib/urldata.h]: Probably also needs `USE_WINDOWS_SSPI` like in the other conditions?

**Comment by MarcelRaad** [comment]
Yes, looks unrelated. The code changes look good to me too,

File: lib/urldata.h (hunk 1 above)
Comment (georgeok) [reply lib/urldata.h]: Thanks for the feedback. Code updated!

**Comment by MarcelRaad** [comment]
Testing right now. I don't expect any problems, so if there are no objections, I'll merge this later today.

**Comment by MarcelRaad** [comment]
Thank you very much!

**Pull request closed without merging**